pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -196,10 +223,133 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 			return reconcile.Result{}, err
 		}
 	}
+	addressSpaceURL := ""
+	//Begin AMQ resource reconcile
+	if instance.Spec.UseMessageBroker {
+		//check that address space exists
+		foundAddressSpace := newAddressSpace(instance)
+
+		// Set UnifiedPushServer instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, foundAddressSpace, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: foundAddressSpace.Name, Namespace: foundAddressSpace.Namespace}, foundAddressSpace)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Address Space", "AddressSpace.Namespace", foundAddressSpace.Namespace, "AddressSpace.Name", foundAddressSpace.Name)
+			err = r.client.Create(context.TODO(), foundAddressSpace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{Requeue: true}, nil
+		} else if err != nil {
+			return reconcile.Result{}, err
+		} else if !foundAddressSpace.Status.IsReady {
+			return reconcile.Result{Requeue: true}, nil
+		} else {
+			reqLogger.Info("Found AddressSpace for UPS")
+		}
+
+		/*Address space exists and is ready*/
+
+		for _, status := range foundAddressSpace.Status.EndpointStatus {
+			if status.Name == "messaging" { //magic value
```

Comment:
Could we ever get to the end of this range loop without having set a value, or will there always be an EndpointStatus called messaging?

Reply:
There will always be one called messaging, see https://enmasse.io/documentation/0.28.2/openshift/